### ISSUE TITLE
Enhancement: Parse string results

### ISSUE THREAD

**Issue by MrNeocore** [python-pinot-dbapi/pinot-dbapi#9]
**Proposition:** Enhancement

In some cases, the Pinot Broker REST API returns string values [e.g. "Infinity"] as results (as part of `rows`) when the value cannot be represented as a JSON "number".

To improve predictability from the Python client, should we parse this output string(s) as a Python native numeric type ?

Simply parsing with `float` should cover many cases and allow for more "natural" handling of Pinot's results from Python.

Should we standardize this approach for other types ?

**Comment by xiangfu0** [comment]
Agreed. We should handle this case base on the python type and value.

**Comment by MrNeocore** [comment]
@xiangfu0 Alright, I'll have a deeper look at the code and ready a PR !
